### ISSUE TITLE
'parent' keyword misused

### ISSUE THREAD

**Issue by s1lver** [kalessil/phpinspectionsea#1160]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug/False-Positive                          |
| Plugin         | Php Inspections (EA Extended)  |
| Language level | PHP 7.2                       |

I used Yii2 framework. I get a lot of false positives.

![default](https://user-images.githubusercontent.com/4567634/50907923-98289a80-1439-11e9-8849-fff31311757f.png)

```php
class ActiveRecord extends BaseActiveRecord
{
    /**
     * {@inheritdoc}
     * @return ActiveQuery the newly created [[ActiveQuery]] instance.
     */
    public static function find()
    {
        return Yii::createObject(ActiveQuery::className(), [get_called_class()]);
    }
}

class Test extends ActiveRecord
{
    ....
}

class TestSearch extends Test 
{
    public function search($params)
    {
        $query = parent::find();//It was probably intended to use '$this->find()' here.
    }
}
```

**Comment by orklah** [comment]
For static methods, we should suggest self:: instead of $this->

**Comment by orklah** [comment]
For me it's the same principle that $this->

parent:: has a different meaning only if the method is redefined in the current class or a child class. For every other cases, it's redundant to call directly the parent:: because every method is inherited in the current class.

It can be on purpose as it will always call the part when someone redefine the method but if it's not explicitly the point, I think it's a misuse of the parent:: keyword

**Comment by kalessil** [comment]
`$this` should not be suggested for sure, thank you @s1lver for reporting. I also think `self::` is a great idea, even if wasn't the part of the original plan here.

**Comment by kalessil** [comment]
Fixed!
